ZAPD/BinaryWriter.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "Stream.h"
+#include <memory>
+#include <array>
+#include <vector>
+#include <string>
+
+class BinaryWriter
```

Comment:
Same here, why are this classes needed?
Maybe we could move classes like this one and `MemoryStream`, and some others already in the repo like `OutputFormatter`, `ImageBackend`, `StringHelper`, Directory and so on to a folder `ZAPD/Utils/` or something. What do you think?

Reply:
These classes help with reading/writing binary data in a C#-like way. I agree it should probably be put in a utils folder or even made its own library (since these classes would be useful beyond just ZAPD itself).